contracts/remarkables/src/contract.rs

```diff
@@ -0,0 +1,1000 @@
+const CONTRACT_VERSION: &str = env!("CARGO_PKG_VERSION");
+const INSTANTIATE_CW721_REPLY_ID: u64 = 1;
+
+// actions for executing messages
+const ACTION_INSTANTIATE: &str = "instantiate";
+const ACTION_INSTANTIATE_CW721_REPLY: &str = "instantiate_cw721_reply";
+const ACTION_MINT: &str = "mint";
+const ACTION_UPDATE_ADMIN: &str = "update_admin";
+const ACTION_UPDATE_RARITY_MINT_FEES: &str = "update_rarity_mint_fees";
+
+// attributes for executing messages
+const ATTRIBUTE_ACTION: &str = "action";
+const ATTRIBUTE_SENDER: &str = "sender";
+const ATTRIBUTE_ADMIN: &str = "admin";
+const ATTRIBUTE_CW721_CODE_ID: &str = "cw721_code_id";
+const ATTRIBUTE_NEW_ADMIN: &str = "new_admin";
+const ATTRIBUTE_RARITY_LEVEL: &str = "rarity_level";
+const ATTRIBUTE_RECIPIENT: &str = "recipient";
+const ATTRIBUTE_TOKEN_ID: &str = "token_id";
+const ATTRIBUTE_TOKEN_URI: &str = "token_uri";
+
+#[cfg_attr(not(feature = "library"), entry_point)]
+pub fn instantiate(
+    deps: DepsMut<DesmosQuery>,
+    env: Env,
+    info: MessageInfo,
+    msg: InstantiateMsg,
+) -> Result<Response<DesmosMsg>, ContractError> {
+    msg.validate()?;
+    // Save the config
+    let admin_addr = deps.api.addr_validate(&msg.admin)?;
+    CONFIG.save(
+        deps.storage,
+        &ConfigState {
+            admin: admin_addr,
+            subspace_id: msg.subspace_id.into(),
+            cw721_code_id: msg.cw721_code_id.into(),
+        },
+    )?;
+    // Save the info of rarities
+    RARITIES.save(deps.storage, &msg.rarities)?;
+    let subspace_id = msg.subspace_id.u64();
+    // Check subspace exists and it is owned by the sender.
+    let subspace = SubspacesQuerier::new(deps.querier.deref())
+        .query_subspace(subspace_id)
+        .map_err(|_| ContractError::SubspaceNotFound { id: subspace_id })?
+        .subspace;
+    if info.sender != subspace.owner {
+        return Err(ContractError::NotSubspaceOwner {
+            caller: info.sender,
+        });
+    }
+    set_contract_version(deps.storage, CONTRACT_NAME, CONTRACT_VERSION)?;
+    // Submessage to instantiate cw721 contract
+    let cw721_submessage = SubMsg::reply_on_success(
+        wasm_instantiate(
+            msg.cw721_code_id.into(),
+            &Cw721InstantiateMsg {
+                name: msg.cw721_instantiate_msg.name,
+                symbol: msg.cw721_instantiate_msg.symbol,
+                minter: env.contract.address.to_string(),
+            },
+            info.funds,
+            "remarkables_cw721".to_string(),
+        )?,
+        INSTANTIATE_CW721_REPLY_ID,
+    );
+    Ok(Response::new()
+        .add_attribute(ATTRIBUTE_ACTION, ACTION_INSTANTIATE)
+        .add_attribute(ATTRIBUTE_SENDER, &info.sender)
+        .add_attribute(ATTRIBUTE_ADMIN, msg.admin)
+        .add_attribute(ATTRIBUTE_CW721_CODE_ID, msg.cw721_code_id)
+        .add_submessage(cw721_submessage))
+}
+
+#[cfg_attr(not(feature = "library"), entry_point)]
+pub fn execute(
```

Comment:
@dadamu Since I would like to make possible in the future to burn Remarkables to obtain in-app advantages, could you check if, once burned, the token-id can be re-minted? I've checked the `Burn` method and it looks like it's not possible but I would like to have your confirmation on this

Reply:
It is possible, cw721 `Burn` method removed the token-id in its state.

Reply:
Disallowed the behavior to mint burned token post again.